fe/src/main/java/org/apache/doris/transaction/GlobalTransactionMgr.java

```diff
@@ -447,6 +447,115 @@ public void commitTransaction(long dbId, long transactionId, List<TabletCommitIn
         // 6. update nextVersion because of the failure of persistent transaction resulting in error version
         updateCatalogAfterCommitted(transactionState, db);
         LOG.info("transaction:[{}] successfully committed", transactionState);
+
+        // 7. try publishing version as soon as possible
+        if (couldPublishTransaction(transactionState)) {
+            publishTransaction(transactionState, totalInvolvedBackends);
+        } else {
+            // add publish version tasks. set task to null as a placeholder.
+            // tasks will be created when publishing version
+            for (long backendId : totalInvolvedBackends) {
+                transactionState.addPublishVersionTask(backendId, null);
+            }
+        }
+    }
+
+    // the publishing transaction should meet:
+    // 1. the transaction state is COMMITTED
+    // 2. the partition's visible version plus one should equal transaction version.
+    public boolean couldPublishTransaction(TransactionState transactionState) {
+        Database db = catalog.getDb(transactionState.getDbId());
+        if (null == db) {
+            return false;
+        }
+
+        boolean meetPublishPredicate = true;
+        db.readLock();
+        try {
+            readLock();
+            try {
+                for (TableCommitInfo tableCommitInfo : transactionState.getIdToTableCommitInfos().values()) {
+                    OlapTable table = (OlapTable) db.getTable(tableCommitInfo.getTableId());
+                    if (null == table) {
+                        LOG.warn("table {} is dropped after commit, ignore this table",
+                                tableCommitInfo.getTableId());
+                        continue;
+                    }
+                    for (PartitionCommitInfo partitionCommitInfo : tableCommitInfo.getIdToPartitionCommitInfo().values()) {
+                        Partition partition = table.getPartition(partitionCommitInfo.getPartitionId());
+                        if (null == partition) {
+                            LOG.warn("partition {} is dropped after commit, ignore this partition",
+                                    partitionCommitInfo.getPartitionId());
+                            continue;
+                        }
+                        if (partitionCommitInfo.getVersion() != partition.getVisibleVersion() + 1) {
+                            LOG.info("transactionState {} version don't meet the publish condition", transactionState);
```

Comment:
I doubt there will be too many logs print here when concurrency is high.

Reply:
OK. it's my debug log, I will remove it.